vllm/worker/worker.py

```diff
@@ -125,6 +126,207 @@ def warm_up_model(self) -> None:
+            _pad_to_max(block_table, max_num_blocks_per_seq, pad=0)
+            for block_table in generation_block_tables
+        ]
+
+        # Convert to tensors.
+        tokens_tensor = torch.tensor(padded_input_tokens,
+                                     dtype=torch.long,
+                                     device="cuda")
+        positions_tensor = torch.tensor(padded_input_positions,
+                                        dtype=torch.long,
+                                        device="cuda")
+        slot_mapping_tensor = torch.tensor(padded_slot_mapping,
+                                           dtype=torch.long,
+                                           device="cpu")
+        context_lens_tensor = torch.tensor(context_lens,
+                                           dtype=torch.int,
+                                           device="cpu")
+        selected_token_indices = torch.tensor(selected_token_indices,
+                                              dtype=torch.long,
+                                              device="cuda")
+        categorized_sample_indices = {
+            t: torch.tensor(seq_ids, dtype=torch.int, device="cuda")
+            for t, seq_ids in categorized_sample_indices.items()
+        }
+        block_tables_tensor = torch.tensor(padded_block_tables,
+                                           dtype=torch.int,
+                                           device="cpu")
+
+        seq_data: Dict[int, SequenceData] = {}
+        for seq_group_metadata in seq_group_metadata_list:
+            seq_data.update(seq_group_metadata.seq_data)
+
+        input_metadata = InputMetadata(
+            seq_groups=seq_groups,
+            seq_data=seq_data,
+            prompt_lens=prompt_lens,
+            slot_mapping=slot_mapping_tensor,
+            context_lens=context_lens_tensor,
+            max_context_len=max_context_len,
+            block_tables=block_tables_tensor,
+            selected_token_indices=selected_token_indices,
+            categorized_sample_indices=categorized_sample_indices,
+            sliding_window=self.sliding_window,
+        )
+
+        # Create attention mask
+        if max_num_blocks_per_seq != 0:
+            attn_masks = torch.zeros((max_num_blocks_per_seq, len(input_tokens), self.block_size), dtype=torch.int64)
+            for i in range(0, max_num_blocks_per_seq):
+                for seq_id in range(len(input_tokens)):
+                    if (i * self.block_size) < context_lens[seq_id] and (i + 1) * self.block_size > context_lens[seq_id]:
+                        attn_masks[i][seq_id, :context_lens[seq_id] % self.block_size] = 1
+                    elif (i + 1) * self.block_size <= context_lens[seq_id]:
+                        attn_masks[i][seq_id, :] = 1
+            input_metadata.attention_masks = attn_masks.to(device="cuda")
+        print("input token shape: ", tokens_tensor.shape)
+        return tokens_tensor, positions_tensor, input_metadata
+
```

Comment:
this wasn't added by us, nor does it seem to add any hpu-related functionality - why do we have it here?